File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Tanimlamalar/Departments/DepartmentsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace MuayeneYonetimPortali.Tanimlamalar.Columns;

[ColumnsScript("Tanimlamalar.Departments")]
[BasedOnRow(typeof(DepartmentsRow), CheckNames = true)]
public class DepartmentsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int DepartmentId { get; set; }
    [EditLink]
    public string Name { get; set; }
}

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Tanimlamalar/DepartmentsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DepartmentsRow } from "./DepartmentsRow";

export interface DepartmentsColumns {
    DepartmentId: Column<DepartmentsRow>;
    Name: Column<DepartmentsRow>;
}

export class DepartmentsColumns extends ColumnsBase<DepartmentsRow> {
    static readonly columnsKey = 'Tanimlamalar.Departments';
    static readonly Fields = fieldsProxy<DepartmentsColumns>();
}